confgen/api_internal.go

```diff
@@ -966,3 +1002,85 @@ func fetchVolumeInfo(confMap conf.ConfMap) (whoAmI string, localVolumeGroupMap v
 
 	return
 }
+
+func IsVolumeSharedSMB(confMap conf.ConfMap, volumeName string) (shared bool, err error) {
+
+	volumeSection := "Volume:" + volumeName
+	shareName, err := fetchStringSet(confMap, volumeSection, "SMBShareName", nil)
+	if err != nil {
+		return
+	}
+
+	if shareName != "" {
+		shared = true
+	}
+	return
+
+}
+
+func IsVolumeSharedNFS(confMap conf.ConfMap, volumeName string) (shared bool, err error) {
+
+	volumeSection := "Volume:" + volumeName
+	clientList, err := fetchStringSet(confMap, volumeSection, "NFSExportClientMapList", nil)
+	if err != nil {
+		return
+	}
+
+	if clientList != "" {
+		shared = true
+	}
+	return
+
+}
+
+// IsVolumeGroupSharedSMB Returns true if any volume in the volume group is
+// shared using SMB.  While we don't support having a volume be shared by both
+// SMB and NFS, it seems like two different volumes in a volume group could be,
+// so a volume group might be "shared" using both protocols.
+//
+func IsVolumeGroupSharedSMB(confMap conf.ConfMap, vgName string) (shared bool, err error) {
```

Comment:
I see here what is going on... but why is this called "Shared"? I get that the NFS form is...

Reply:
Are you saying these should be renamed **IsVolumeGroupSMB()** and **IsVolumeGroupNFS()**?

Reply:
I don't know.  Bob's `IsVolumeSMB()` seems strange to me ... i don't think of a volume group being "SMB".

i could use the more verbose `IsVolumeGroupSharedViaSMB()`/`IsVolumeGroupSharedViaNFS()` i suppose ...